static/js/rendered_markdown.js

```diff
@@ -235,6 +235,23 @@ export const update_elements = (content) => {
         });
     });
 
+    // We display the still image normally but show the GIF on hover.
+    content.find("a[href*='giphy.com/media']").each(function () {
+        const $a_gif = $(this);
+        const gif_url = $a_gif.attr("href");
+        // Add `_s` to the gif file name to get the still image.
+        const still_image_url = gif_url.replace("giphy.gif", "giphy_s.gif");
```

Comment:
Hmm, does this replacement algorithm work with camo?

Reply:
How can I test it? I haven't worked with camo before.

Reply:
It doesn't work with `camo`, sadly.  `camo` means that we rewrite external URLs into things like `/external_content/45825b103b11eec001b07e4cda4ccf8ad2ca9362/68747470733a2f2f6d65646961332e67697068792e636f6d2f6d656469612f3554326546714a4430306f53492f67697068792e6769663f6369643d6336323363623335327963687863707a397561346d736f67626133656f6c6c38766e686c30777272326d33757a347135267269643d67697068792e6769662663743d67` where the URL is encoded into the part after the last slash.  You can un-hex it to (`68747470` -> `http` at the start, for example) get the original URL.

I don't think you can test in dev without [downloading and running `camo`](https://github.com/cactus/go-camo/releases) and setting `CAMO_URI` / `CAMO_KEY`.

Reply:
To be clear, you can potentially look for the string `67697068792e636f6d2f6d65646961` (`giphy.com/media` hex-encoded) in the URL, and `gif_url.replace("67697068792e676966","67697068795f732e676966")`.

Reply:
A camo URL can’t just be string-replaced; it’s cryptographically signed with a key that the client isn’t allowed to know. It needs to be rebuilt on the server with `get_camo_url`.

Reply:
Gah, of course.  Yeah, so this implies that the server side needs to be more aware of giphy in some form, to be able to provide both URLs to the client.